src/main/java/teammates/ui/controller/FeedbackSubmissionEditSaveAction.java

```diff
@@ -367,6 +419,63 @@ private FeedbackResponseAttributes extractFeedbackResponseData(
         return response;
     }
 
+    private void extractFeedbackParticipantCommentsData(FeedbackQuestionAttributes questionAttributes,
+            int questionIndex, FeedbackResponseAttributes response, int responseIndex) {
+        if (!questionAttributes.getQuestionDetails().isFeedbackParticipantCommentsOnResponsesAllowed()) {
+            return;
+        }
+        String commentText;
+        String commentId =
```

Comment:
The way existing comments feature is implemented, this commentId helps in mapping itself to responses. Changing this would lead to different implementations for student and instructor side. I think it would be better to change how we implement indexes of comment under another PR as you said earlier?

Reply:
> I think it would be better to change how we implement indexes of comment under another PR as you said earlier

Better do it in another PR.

We can stick to the implementation of instructor side and reuse the code (only if we can reuse the code). If the reuse of existing bring more confusion (many if-else), it would be better to just write a new one (I think the situation is already there for comment row).